### PR TITLE
feat: swap contract address search support

### PR DESCRIPTION
**Pull request by magiziz**
Fixes BX-1330

## What changed (plus any additional context for devs)

- Swap page now supports contract address search. It'll also search on all supported networks instead of single choosen network.

## Screen recordings / screenshots

https://github.com/rainbow-me/browser-extension/assets/53529533/2f61def6-9a65-4fda-9832-cfee73c18ab4

## What to test
- Go to swap page
- Paste in any token contract address and make sure it searches the token on all supported networks. For example `0x50c5725949a6f0c72e6c4a641f24049a917db0cb` should search for DAI on base and LYRA on optimism.
image
- Make sure you don't see "Filter by network" dropdown option because we search for all networks.
- Make sure the swap flow works the same as usual.

### CODE REVIEW

File: src/core/resources/search/tokenSearch.ts [hunk 1]
```diff
@@ -202,13 +219,14 @@ export function useTokenSearchAllNetworks({
         }),
         queryFn: tokenSearchQueryFunction,
         select: (data: SearchAsset[]) => {
-          if (!isAddress(query)) return [];
+          if (!isAddress(query) || testnetMode) return [];
```
Comment (estebanmino) [inline src/core/resources/search/tokenSearch.ts:222]: there's no token search in testnets?

Comment (magiziz) [reply src/core/resources/search/tokenSearch.ts:222]: Nope :/

Comment (estebanmino) [reply src/core/resources/search/tokenSearch.ts:222]: ok cool

**Review by estebanmino — APPROVED** [review]
lgtm and is working as expected 🚀

**Review by derHowie — APPROVED** [review]
search looked good to me 🫡

**Pull request merged**